solidity-plugin/src/main/java/org/sonarsource/solidity/MetricsVisitor.java

```diff
@@ -0,0 +1,189 @@
+import org.sonarsource.solidity.frontend.SolidityParser.DoWhileStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.EmitStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ExpressionStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ForStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.FunctionDefinitionContext;
+import org.sonarsource.solidity.frontend.SolidityParser.IfStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ReturnStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ThrowStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.VariableDeclarationStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.WhileStatementContext;
+
+public class MetricsVisitor extends SolidityBaseVisitor<Token> {
+  protected FileMeasures fileMeasures;
+  int functionCounter = 0;
+  int statements = 0;
+  int contractCounter = 0;
+
+  public MetricsVisitor(SolidityParser parser) {
+    this.fileMeasures = new FileMeasures();
+    parser.sourceUnit().accept(this);
+    this.fileMeasures.setStatementNumber(statements);
+    this.fileMeasures.setFunctionNumber(functionCounter);
+    this.fileMeasures.setContractNumber(contractCounter);
+    this.fileMeasures.setCommentLinesNumber(computeLinesOfComments(parser.comments));
+    TerminalNode node = parser.sourceUnit().EOF();
+    this.fileMeasures.setLinesOfCodeNumber(node.getSymbol().getLine());
+  }
+
+  private int computeLinesOfComments(Set<Token> tokens) {
+    int comments = 0;
+    for (Token token : tokens) {
+      if (token.getType() == 118) {
+        comments++;
+      } else {
+        comments += token.getText().split("\r\n|\r|\n").length;
+      }
+    }
+    return comments;
+  }
+
+  // @Override
+  // public Token visitSourceUnit(SourceUnitContext ctx) {
+  // List<FunctionDefinitionContext> functionList = contractPartList
+  // .stream()
+  // .map(x -> x.functionDefinition())
+  // .filter(Objects::nonNull)
+  // .collect(Collectors.toList());
+  //
+  // fileMeasures.setFunctionNumber(functionList.size());
+  //
+  // long statementsCounter = 0;
+  // List<StatementContext> statementList = new ArrayList<>();
+  //
+  // functionList
+  // .stream()
+  // .map(FunctionDefinitionContext::block)
+  // .map(BlockContext::statement)
+  // .filter(Objects::nonNull)
+  // .forEach(x -> statementList.addAll(x));
+  //
+  // BlockVisitorClass vis = new BlockVisitorClass();
+  // functionList
+  // .stream()
+  // .map(x -> x.accept(vis))
+  // .forEach(x -> System.out.println(x));
+  // return super.visitSourceUnit(ctx);
+  // }
+
+  @Override
+  public Token visitContractDefinition(ContractDefinitionContext ctx) {
+    contractCounter++;
+    return super.visitContractDefinition(ctx);
+  }
+
+  @Override
+  public Token visitContractPart(ContractPartContext ctx) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this method to simply inherit it. ([squid:S1185](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1185))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=16&issues=AWQTGK7hbXXTsx7O4tIO&open=AWQTGK7hbXXTsx7O4tIO)

[ISSUE_KEY]: AWQTGK7hbXXTsx7O4tIO